forta-uniswap-event-py/src/agent_test.py

```diff
@@ -0,0 +1,208 @@
+import json
+import pytest
+import os
+
+from web3 import Web3
+
+from forta_agent import Finding, FindingSeverity, FindingType, create_transaction_event
+from agent import handle_transaction, AttrDict
+
+
+BURN_ADDR = "0x000000000000000000000000000000000000dEaD"
+
+
+@pytest.fixture
+def alert():
+    """
+    Various properties of the alert that is raised inside agent.py
+    """
+    alert_large_swap = Finding(
+        {
+            "name": "Uniswap V2 swap detector",
+            "description": "Large swap on Uniswap V2 detected",
+            "alert_id": "AE-UNISWAP-LARGESWAP-EVENT",
+            "type": FindingType.Suspicious,
+            "severity": FindingSeverity.Low,
+        }
+    )
+
+    return alert_large_swap
+
+@pytest.fixture
+def uniswap_v2_router_addr():
+    """
+    Load the configureable Uniswap V2 router address.
+    """
+    dirname = os.path.dirname(os.path.abspath(__file__))
+    config_file = os.path.join(dirname, 'config', 'agent-settings.json')
+
+    with open(config_file, 'r') as f:
+        data = json.loads(f.read())
+
+    return Web3.toChecksumAddress(data['uniswap_v2_router_addr'])
+
+
+def check_alerts(expected_alert, found_alert):
+    """
+    Compares two Finding type objects and asserts that all fields are the same.
+    Raises AssertionError if a difference is found between the two alerts
+    """
+    assert expected_alert.name == found_alert.name
+    assert expected_alert.description == found_alert.description
+    assert expected_alert.alert_id == found_alert.alert_id
+    assert expected_alert.severity.value == found_alert.severity.value
+    assert expected_alert.type.value == found_alert.type.value
+
+
+def gen_tx_data(value="0", from_=BURN_ADDR, to=BURN_ADDR, data="0x"):
+    """
+    Generate a dict containing transaction data to be used in mocking a transaction
+    """
+    transaction_data = {}
+    transaction_data["value"] = value
+    transaction_data["from_"] = from_
+    transaction_data["to"] = to
+    transaction_data["data"] = data
+
+    return {"transaction": transaction_data}
+
+
+def gen_tx_receipt(event=None):
+    """
+    Generate a transaction receipt to be used in creating a mock transaction
+    """
+    logs = []
+    if event == "deposit":
+        logs.append(gen_log_receipt("deposit"))
+
+    elif event == "withdrawal":
+        logs.append(gen_log_receipt("withdrawal"))
+
+    elif event == "swap":
+        logs.append(gen_log_receipt("swap"))
+
+    temp_dict = AttrDict(
+        {
+            "status": True,
+            "root": "",
+            "cumulative_gas_used": 0,
+            "gas_used": 0,
+            "logs_bloom": "0x0",
+            "logs": logs,
+            "contract_address": None,
+            "block_hash": "0x85d8e4f37fd7146d82d3fdb851668ad88eb5351cd535c5fc3d62a40eed817c92",
```

Comment:
Any reason why block_hash and block_number look like real data, but everything else is zero/empty/None?

Reply:
No reason, I nulled out those two fields to reduce confusion